storage/innobase/fsp/fsp0file.cc

```diff
@@ -346,6 +346,13 @@ Datafile::read_first_page(bool read_only_mode)
 		if (!fsp_flags_is_valid(m_flags, m_space_id)) {
 			ulint cflags = fsp_flags_convert_from_101(m_flags);
 			if (cflags == ULINT_UNDEFINED) {
+				if (FSP_FLAGS_GET_PAGE_SSIZE(m_flags) == 2) { 
+					/* MySQL-8.0 SDI flag */
+					ib::error() << "MySQL-8.0 tablespace in "
+						<< m_filepath << " detected."
+						" Restart in MySQL-8.0 for migration/recovery.";
```

Comment:
I think that we should prefer `sql_print_error("InnoDB: …");` for messages that do not really depend on any fancy `operator<<` handling.